File: aplication/client/FE/src/app/shared/model/entity/Comment.ts
import {Movie} from './Movie';
import {Account} from './Account';
import {JsonProperty} from 'jsog-typescript';

export class Comment {
  id: number;
  content: string;
  seen: number;
  movie: Movie;
  account: Account;


  constructor(id: number, content: string, seen: number, movie: Movie, account: Account) {
    this.id = id;
    this.content = content;
    this.seen = seen;
    this.movie = movie;
    this.account = account;
  }
}

File: aplication/server/BE/src/main/java/com/model/entity/Comment.java
package com.model.entity;

import com.voodoodyne.jackson.jsog.JSOGGenerator;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import org.hibernate.annotations.Type;

import javax.persistence.*;

@Entity
@JsonIdentityInfo(generator= JSOGGenerator.class)
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String content;

    @Type(type = "org.hibernate.type.NumericBooleanType")
    private Integer rate;
    @ManyToOne
    @JoinColumn(name = "movie_id", referencedColumnName = "id")
    private Movie movie;

    @ManyToOne
    @JoinColumn(name = "account_id", referencedColumnName = "id")
    private Account account;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Movie getMovie() {
        return movie;
    }

    public void setMovie(Movie movie) {
        this.movie = movie;
    }

    public Account getAccount() {
        return account;
    }

    public void setAccount(Account account) {
        this.account = account;
    }

    public Integer getRate() {
        return rate;
    }

    public void setRate(Integer rate) {
        this.rate = rate;
    }
}
